plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,321 @@
+        }
+    }
+
+    @Override
+    protected void renameTable(ConnectorSession session, String catalogName, String schemaName, String tableName,
+            SchemaTableName newTable)
+    {
+        String sql = format("RENAME TABLE %s.%s TO %s.%s", schemaName, tableName, newTable.getSchemaName(),
+                newTable.getTableName());
+        execute(session, sql);
+    }
+
+    @Override
+    public boolean isLimitGuaranteed(ConnectorSession session)
+    {
+        return true;
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection,
+            JdbcTypeHandle typeHandle)
+    {
+        String jdbcTypeName = typeHandle.getJdbcTypeName()
+                .orElseThrow(() -> new TrinoException(JDBC_ERROR, "Type name is missing: " + typeHandle));
+
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+        if (jdbcTypeName.equals("UInt8")) {
+            return Optional.of(booleanColumnMapping());
+        }
+        if (jdbcTypeName.equals("Int8")) {
+            return Optional.of(tinyintColumnMapping());
+        }
+        if (jdbcTypeName.equals("Int16")) {
+            return Optional.of(smallintColumnMapping());
+        }
+        if (jdbcTypeName.equals("Int32")) {
+            return Optional.of(integerColumnMapping());
+        }
+        if (jdbcTypeName.equals("Int64") || jdbcTypeName.equals("bigint")) {
+            return Optional.of(bigintColumnMapping());
+        }
+        if (jdbcTypeName.equals("Float32")) {
+            return Optional.of(realColumnMapping());
+        }
+        if (jdbcTypeName.equals("Float64")) {
+            return Optional.of(doubleColumnMapping());
+        }
+        if (typeHandle.getJdbcType() == Types.DECIMAL) {
+            int precision = min(typeHandle.getColumnSize().orElse(0), Decimals.MAX_PRECISION);
+            int decimalDigits = typeHandle.getDecimalDigits()
+                    .orElseThrow(() -> new IllegalStateException("decimal digits not present"));
+            int scale = min(decimalDigits, getDecimalDefaultScale(session));
+            return Optional.of(decimalColumnMapping(createDecimalType(precision, scale),
+                    getDecimalRoundingMode(session)));
+        }
+        if (jdbcTypeName.equals("String")) {
+            return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+        }
+        if (jdbcTypeName.equals("Date")) {
+            return Optional.of(dateColumnMapping());
+        }
+        if (jdbcTypeName.equals("DateTime")) {
+            // clickhouse not implemented for type=class java.time.LocalDateTime
+            // TODO replace it using timestamp relative function after clickhouse adds support for LocalDateTime
+            return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
```

Comment:
DateTime probably should be mapped to timestamp.

please leave it out from this PR, and make into a follow up. timestamps often require multiple rounds of review because they are often hard to get right

Reply:
ok